include/mips-emulator/instruction.hpp

```diff
@@ -138,14 +139,30 @@ namespace mips_emulator {
 
         // Floating Point Format field
         enum class FPURTypeOp : uint8_t {
-            e_fmt_s = 16,  // Use Single
-            e_fmt_d = 17,  // Use Double
-            e_fmt_w = 20,  // Use 32 bit
-            e_fmt_l = 21,  // Use 64 bit
+            e_fmt_s = 16, // Use Single
+            e_fmt_d = 17, // Use Double
+            e_fmt_w = 20, // Use 32 bit
+            e_fmt_l = 21, // Use 64 bit
             e_cmp_condn_s = 24,
             e_cmp_condn_d = 25,
         };
 
+        // Opcode for special3 instruction
+        enum class Special3Opcode : uint8_t {
+            e_special3 = 0b011111,
+        };
```

Comment:
Does this need it's own enum class? It might be better to do this the same way that the special0 opcode is handled for regular instructions and just set it to 0b011111 in the constructor. (I might have missed something that makes this approach better)